### ISSUE TITLE
CodedBufferReader should have a getter for _bufferPos

### ISSUE THREAD

**Issue by jnstahl** [google/protobuf.dart#562]
When reading delimited messsages from a socket, the data may come in in smaller packets.
The typical way to read the messages is by first decoding the message size from with readInt32, then it is possible to decide if the entire message is already in the buffer or if it is required to wait for additional packages before deoding the message.

However as the message size is a varint, the number of bytes used for the size descriptor may vary. There is currently no way of telling how many bytes the CodedBufferReader has consumed in the readInt32, as the _bufferPos member is private.

Adding a getter to access this property would make decoding of delimited message much easier. The coding for this is trivial.

**Comment by jnstahl** [comment]
Alternatively there could be remaining() function which returns the number of bytes left in the source buffer:

`int remaining() => _currentLimit - _bufferPos;`

**Comment by osa1** [comment]
`CodedBufferReader.readInt32` is probably public just to allow testing. I'm not sure if it's something we want to export as a part of the public API. We should probably add `@visibleForTesting` to `readInt32` and other methods that we only make public for testing. Any thoughts on this @sigurdm?

I think it should be possible implement an incremental binary deserializer in a separate library.

In the meantime I suggest copying the `readInt32` implementation and modifying it.

**Comment by sigurdm** [comment]
Yeah I don't think the CodedBufferReader should ever have been exposed.

Maybe we make it private and have a typedef alias to it that is `@Deprecated`, that can then be removed at some point.
